### PR TITLE
Ardupilot gio

### PR DESCRIPTION
**Pull request by GiovannaAF**

### CODE REVIEW

**Comment by Gone4Dirt** [comment]
Thanks for the PR Giovanna.  I would suggest that you wait until this PR:  #18044 has been added as it will make adding your extra RPM instances easier.

A couple of things to note when creating a PR to ArduPilot.  You need to make sure that your commit history is tidy.  So typically, you do a commit per functional change, per library or vehicle.  See the wiki page linked below for some guidance on preparing your PR in general.  Also have a look at some other PRs to the ArduPilot repo for examples.
https://ardupilot.org/dev/docs/submitting-patches-back-to-master.html

You also need to make sure that your code is rebased on master so that it is up to date and there are no conflicting patches that have been applied since the commit that you started writting your code.

File: libraries/AP_RPM/AP_RPM.h [hunk 1]
```diff
@@ -99,7 +99,7 @@ class AP_RPM
 
     RPM_State state[RPM_MAX_INSTANCES];
     AP_RPM_Backend *drivers[RPM_MAX_INSTANCES];
-    uint8_t num_instances:2;
+    uint8_t num_instances:3;  // This is a bit field not =
```
Comment (magicrub) [inline libraries/AP_RPM/AP_RPM.h:102]:
```suggestion
    uint8_t num_instances;
```
Seems you already tripped on this. Why not just make it a full byte to save everyone the headache?

**Comment by GiovannaAF** [comment]
> @GiovannaAF Hello, do you indent to work on this ? on rebase it will just change the max number of RPM instance to 4

Yes, sure.  But I I don't know how to rebase the code.

**Comment by peterbarker** [comment]
Thanks for the PR, @GiovannaAF .  ArduPilot master has support for four RPM instances now, so I'm going to close this one.

Also, our Wiki has information about rebasing code if you haven't gotten to working that sutff out yet :-)

**Pull request closed without merging**